### PR TITLE
feat: replace some of the hardcoded dpos params by paramas from confi…

### PR DESCRIPTION
**Pull request by JakubFornadel**
…g + config cleanup

### CODE REVIEW

File: taraxa/state/dpos/precompiled/api.go [hunk 1]
```diff
@@ -18,35 +18,54 @@ type API struct {
...
 type Config = struct {
 	EligibilityBalanceThreshold *big.Int
 	VoteEligibilityBalanceStep  *big.Int
 	MaximumStake                *big.Int
 	MinimumDeposit              *big.Int
 	CommissionChangeDelta       uint16
-	CommissionChangeFrequency   types.BlockNum
-	DepositDelay                types.BlockNum
-	WithdrawalDelay             types.BlockNum
+	CommissionChangeFrequency   uint64 // [number of blocks]
```
Comment (MatusKysel) [inline taraxa/state/dpos/precompiled/api.go:28]: why did you change type? if you keep it as it is you do not need any comment

**Review by MatusKysel — APPROVED** [review]

File: taraxa/state/dpos/precompiled/api.go (hunk 1 above)
Comment (JakubFornadel) [reply taraxa/state/dpos/precompiled/api.go:28]: because it is weird, on every other place this type means exact block number. not number of blocks(counter).... and when you look at it you think automatically it is block number and not counter

Comment (JakubFornadel) [reply taraxa/state/dpos/precompiled/api.go:28]: hmm so how did you know before it was block counter based on types.BlockNum ? This type is saying this is block number

Comment (JakubFornadel) [reply taraxa/state/dpos/precompiled/api.go:28]: pls check the codebase,  types.BlockNum always represent the exact block number so this was totally misleading as when you are used to it, you automatically consider any value stored there to be block number and not some counter...

Comment (MatusKysel) [reply taraxa/state/dpos/precompiled/api.go:28]: maybe I understand it wrongly but variable defines what it is holding e.g. (block) number, (block) range and the type defines of what e.g. block number, block range

Comment (JakubFornadel) [reply taraxa/state/dpos/precompiled/api.go:28]: I dont understand what you meant by that. If you define custom type, which is called BlockNum that means you are using it for variables that hold block number and nothing else, period. If you define variable as string, it holds string - not number of characters in the string, or anything else.... You are just lucky that types.BlockNum is uin64 here so you could use it as block number as well as counter. If it was *big.Int you wouldn't use it as a block counter for delay/frequency because these variables will never be some huge number so why would you use big.Int. As I said, if 
 you check the codebase - this type is used only for variables that hold actual block number. Example:

```
type Delegation struct {
	// Num of delegated tokens == delegator's stake
	Stake *big.Int

	// Block number related to rewards
	LastUpdated types.BlockNum
}
```

How do you know that LastUpdated is the actual block number and not some counter like 200 blocks ago ? Well very easily, because the type is types.BlockNum, which says that this variables stores block number and nothing else. 

So no, the counter should definitely not have type BlockNum

**Review by gaoqi8649 — APPROVED** [review]

File: taraxa/state/dpos/precompiled/api.go (hunk 1 above)
Comment (MatusKysel) [reply taraxa/state/dpos/precompiled/api.go:28]: ok, then let's use smaller types if you want to keep it as uint64 does not make any sense. uint32 should be just fine and we safe a lot of space passing this struct to go
